pkg/provenance.go

```diff
@@ -402,6 +418,31 @@ func VerifyWorkflowIdentity(id *WorkflowIdentity, source string) error {
 	return nil
 }
 
+// Only allow `@refs/heads/main` for the builder and the e2e tests that need to work at HEAD.
+// This lets us use the pre-build builder binary generated during release (release happen at main).
+// For other projects, we only allow semantic versions that map to a release.
+func verifyTrustedBuilderRef(id *WorkflowIdentity, ref string) error {
```

Comment:
This seems like something you'll want a test or two for.

Reply:
the tests exist under provenance_test.go (Test_verifyTrustedBuilderRef).